tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -286,6 +285,49 @@ private static void ExtractActivityLinks<TInner>(Span span, IActivity5? activity
             }
         }
 
+        private static void ExtractActivityEvents<TInner>(Span span, IActivity5? activity5)
+            where TInner : IActivity
+        {
+            if (activity5 is null)
+            {
+                return;
+            }
+
+            foreach (var activityEvent in activity5.Events)
+            {
+                if (!activityEvent.TryDuckCast<ActivityEvent>(out var duckEvent))
+                {
+                    continue;
+                }
+
+                var eventAttributes = new List<KeyValuePair<string, object>>();
+                foreach (var kvp in duckEvent.Tags)
+                {
+                    if (!string.IsNullOrEmpty(kvp.Key) && IsAllowedAtributeType(kvp.Value))
+                    {
+                        if (kvp.Value is Array array)
```

Comment:
This change mirrors the SpanLink attributes logic, but it does change how events are serialized. I recommend reverting this particular part so that we copy over arrays the same as before, and we can make this sort of change as part of a separate PR